packages/manager/apps/pci-kubernetes/src/api/hooks/kubernetes.ts

```diff
@@ -0,0 +1,47 @@
+import { useMutation, useQuery } from '@tanstack/react-query';
+import {
+  getKubernetesCluster,
+  updateKubernetesCluster,
+} from '@/api/data/kubernetes';
+import queryClient from '@/queryClient';
+
+export const useKubernetesCluster = (projectId: string, kubeId: string) =>
+  useQuery({
+    queryKey: ['project', projectId, 'cluster', kubeId],
```

Comment:
I propose `['project', projectId, 'kube', kubeId]`  as key because the call is `project/${projectId}/kube/${kubeId}`